src/scip/concsolver.c

```diff
@@ -500,7 +500,7 @@ SCIP_RETCODE SCIPconcsolverSync(
       SCIPdebugMessage("syncfreq after reading the next syncdata is %g\n", concsolver->syncfreq);
       syncdata = SCIPsyncstoreGetNextSyncdata(syncstore, concsolver->syncdata, concsolver->syncfreq, concsolver->nsyncs, &concsolver->syncdelay);
    }
-
+   SCIPsyncstoreCheckGapAndStop(syncstore,set->limit_gap);
```

Comment:
I think logically this should better go into the method concurrent.c:execConcsolver() after SCIPconcsolverSync(). It would have the same effect and be clearer.

Reply:
Thanks for your comment！
I think this is mainly a syncstore operation, so I would like to left it in the scip.syncstore .
Another reason is that SCIPconcsolverSync() has already been registered with the events manager. If I wanted to write a check and stop function that executes during the concurrent solver's solving process, I would need to write another callback function and register it, which I believe is unnecessary : )
Furthermore, I believe this is just a minor, trivial feature that is simple enough not to warrant overly complex development and design for it.

Reply:
Maybe we misunderstood each other.  I was just suggesting to have
```
   syncstore = SCIPgetSyncstore(scip);
   assert(syncstore != NULL);

   SCIP_CALL( SCIPconcsolverExec(scip->set->concsolvers[SCIPtpiGetThreadNum()]) );
   SCIP_CALL( SCIPconcsolverSync(scip->set->concsolvers[SCIPtpiGetThreadNum()], scip->set) );
   SCIPsyncstoreCheckGapAndStop(syncstore, scip->set->limit_gap);
```
in execConcsolver(), and not in SCIPconcsolverSync().  Or do you think it is also necessary in SCIPsynchronize()?

Reply:
Actually, I understand what you mean. I tried adding this function here, but it will only execute after the concurrent solver has finished, so it cannot serve as an interrupt during the execution of the concurrent solver.

Reply:
Okay, then let's stay with your solution.